src/client/dfuse/pil4dfs/hook.c

```diff
@@ -1186,3 +1187,101 @@ query_all_org_func_addr(void)
 		}
 	}
 }
+
+int
+query_var_addr_size(const void *ref_func_addr, const char *ref_func_name, const char *var_name,
+		    size_t *var_size, char **var_addr)
+{
+	int         fd, i, j, rc;
+	struct stat file_stat;
+	void       *map_start;
+	Elf64_Ehdr *header;
+	Elf64_Shdr *sections;
+	int         strtab_offset  = 0;
+	void       *symb_base_addr = NULL;
+	int         num_sym = 0, sym_rec_size = 0, sym_offset, rec_addr;
+	char       *sym_name;
+	long int    addr_fun = 0;
+	long int    addr_var = 0;
+	Dl_info     dl_info;
+
+	if (ref_func_addr == NULL || ref_func_name == NULL || var_size == NULL || var_addr == NULL)
+		return EINVAL;
+
+	rc = dladdr(ref_func_addr, &dl_info);
+	if (rc == 0)
+		return ENODATA;
+
+	rc = stat(dl_info.dli_fname, &file_stat);
+	if (rc == -1) {
+		D_DEBUG(DB_ANY, "Fail to query stat of file %s", dl_info.dli_fname);
```

Comment:
should these not be D_ERRORs instead? Same for few other D_DEBUG prints before in error paths